### PR TITLE
AP_Param: allow Vector3f defaults

### PR DESCRIPTION
**Pull request by IamPete1**
Came up while discussing https://github.com/ArduPilot/ardupilot/pull/21245. This allows default values to be set so they do not show up as none default when doing a diff using the new defaults param support over FTP. 

This does also remove the code that was setting the vectors to 1. Because this was done after param load if someone had saved a zeros in there params we would now use it where previously we would change it to all ones.

The param diff with and without this patch after a wipe shows only `INS_ACC3SCAL_` ... `X`,`Y` and `Z`. This is because the removed param set in `AP_InertialSensor` did a loop to `get_accel_count()` which on SITL is 2. So the 3rd set of params remained at zero, with this patch there all one.

### CODE REVIEW

File: libraries/AP_Compass/AP_Compass_Backend.cpp [hunk 1]
```diff
@@ -48,10 +48,6 @@ void AP_Compass_Backend::correct_field(Vector3f &mag, uint8_t i)
 {
     Compass::mag_state &state = _compass._state[Compass::StateIndex(i)];
 
-    if (state.diagonals.get().is_zero()) {
```
Comment (IamPete1) [inline libraries/AP_Compass/AP_Compass_Backend.cpp:51]: If a user has got the diagonals all set to 0 in eeprom we will now do the wrong thing. Seems quite unlikely tho.

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 2]
```diff
@@ -861,14 +861,6 @@ AP_InertialSensor::init(uint16_t loop_rate)
         _start_backends();
     }
 
-    // initialise accel scale if need be. This is needed as we can't
-    // give non-zero default values for vectors in AP_Param
-    for (uint8_t i=0; i<get_accel_count(); i++) {
-        if (_accel_scale[i].get().is_zero()) {
```
Comment (IamPete1) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:867]: as above we do now have the possibility to do the wrong thing if all zeros are saved in eeprom.

File: libraries/AP_Param/AP_Param.cpp [hunk 3]
```diff
@@ -1461,6 +1461,11 @@ void AP_Param::setup_object_defaults(const void *object_pointer, const struct Gr
             void *ptr = (void *)(base + group_info[i].offset);
             set_value((enum ap_var_type)type, ptr,
                       get_default_value((const AP_Param *)ptr, &group_info[i].def_value));
+        } else if (type == AP_PARAM_VECTOR3F) {
+            // Single default for all components
+            void *ptr = (void *)(base + group_info[i].offset);
+            const float default_val = get_default_value((const AP_Param *)ptr, &group_info[i].def_value);
```
Comment (IamPete1) [inline libraries/AP_Param/AP_Param.cpp:1467]: may not be worth garbing the default from the override file since all components are the same.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
nice, thanks!

**Pull request merged**